packages/cms-data-sync/src/utils/assets.ts

```diff
@@ -26,6 +27,34 @@ export const checkIfAssetAlreadyExistsInContentful = async (
   }
 };
 
+export type InlineAssetBody = [id: string, fields: CreateAssetProps];
+
+export const createInlineAssets = async (
+  contentfulEnvironment: Environment,
+  inlineAssetBodies: InlineAssetBody[],
+) => {
+  const assets = await Promise.all(
+    inlineAssetBodies.map(async ([id, fields]) => {
+      const isAssetAlreadyInContentful =
+        await checkIfAssetAlreadyExistsInContentful(contentfulEnvironment, id);
+      if (!isAssetAlreadyInContentful) {
+        console.log(`Creating asset with id ${id}.`);
+        return contentfulEnvironment.createAssetWithId(id, fields);
+      }
+      return null;
+    }),
+  );
+
+  await Promise.all(
```

Comment:
I think we need something a little more sophisticated than Promise.all here, first of all we want it to finish all promises in the array right? Which i think already eliminates Promise.all. People generally opt for Promise.allSettled instead, which waits for all promises to either resolve or reject and returns a list of failures.  We still probably want to reject when one fails. 

Alternatively rename this to `createInlineAsset` and move this promise-failure-handling logic to where its used